zerver/webhooks/patreon/view.py

```diff
@@ -0,0 +1,162 @@
+from typing import Any, Callable, Dict, Optional
+
+from django.http import HttpRequest, HttpResponse
+
+from zerver.decorator import webhook_view
+from zerver.lib.exceptions import UnsupportedWebhookEventType
+from zerver.lib.request import REQ, has_request_variables
+from zerver.lib.response import json_success
+from zerver.lib.webhooks.common import (
+    check_send_webhook_message,
+    validate_extract_webhook_http_header,
+)
+from zerver.models import UserProfile
+
+
+# convert fixture names of type members_create -> member:create
+# this is done to match the http header X_PATREON_EVENT we get from the incoming webhook
+# check patreon webhook docs for more info on why we implemented this
+def get_http_headers_from_filename(http_header_key: str) -> Callable[[str], Dict[str, str]]:
+    def fixture_to_headers(filename: str) -> Dict[str, str]:
+        event_type = ":".join(filename.split("_"))
+        return {http_header_key: event_type}
+
+    return fixture_to_headers
+
+
+fixture_to_headers = get_http_headers_from_filename("HTTP_X_PATREON_EVENT")
+
+# called when new patron joined
+def members_create(payload: Dict[str, Any]) -> str:
+
+    # array of dicts
+    # first item contains the patron count
+    included = payload["included"][0]
+
+    # get the patron count
+    patron_count = included["attributes"]["patron_count"]
+
+    # create the body
+    body = f"New patron joined! :tada:\nPatrons in total: {patron_count}"
```

Comment:
The comments here are excessive and should be removed. Also maybe too many newlines. Ideally we'd want this to just be:
```python3
patron_count = included["attributes"]["patron_count"]
body = f"New patron joined! :tada:\nPatrons in total: {patron_count}"
```